### PR TITLE
Add foundation XYgrid (Only the X part here though)

### PR DESCRIPTION
**Pull request by MrEssex**

### CODE REVIEW

File: src/Layout/XyGrid/XyCell.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+{
+  use BemComponentTrait;
+  use ComponentTrait;
+
+  /** @var string */
+  public const DEVICE_DESKTOP = 'large';
+  /** @var string */
+  public const DEVICE_TABLET = 'medium';
+  /** @var string */
+  public const DEVICE_PHONE = 'small';
```
Comment (bajb) [inline src/Layout/XyGrid/XyCell.php:24]:
```suggestion
```

File: src/Layout/XyGrid/res/xy-grid.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
+@import "~foundation-sites/scss/foundation";
+@import "../../../Foundation/res/foundation";
+
+$container-size: 1140px !default; // The size of the container
```
Comment (bajb) [inline src/Layout/XyGrid/res/xy-grid.scss:4]: max

File: src/Layout/XyGrid/XyGrid.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  /**
+   * @return $this
+   */
+  public function marginY()
+  {
+    $this->addClass('grid-margin-y');
+    return $this;
+  }
+
```
Comment (bajb) [inline src/Layout/XyGrid/XyGrid.php:55]: marginXY

File: src/Layout/XyGrid/XyContainer.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  /**
+   * @return string
+   */
+  public function getBlockName(): string
+  {
+    return 'grid-container';
+  }
+
```
Comment (bajb) [inline src/Layout/XyGrid/XyContainer.php:37]: fluid

File: src/Layout/XyGrid/XyCell.php [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+   * @throws Exception
+   */
+  public function setSizes(XySize ...$sizes)
+  {
+    foreach($sizes as $size)
+    {
+      $getSize = $size->getSize();
+      $offsetSize = $size->getOffset();
+
+      if(($getSize && $offsetSize) && $getSize + $offsetSize > 12)
```
Comment (TomK) [inline src/Layout/XyGrid/XyCell.php:57]:
```suggestion
      if($getSize + $offsetSize > 12)
```

**Review by TomK — APPROVED** [review]

**Pull request merged**